cirq-google/cirq_google/devices/grid_device.py

```diff
@@ -373,6 +461,103 @@ def from_proto(cls, proto: v2.device_pb2.DeviceSpecification) -> 'GridDevice':
 
         return GridDevice(metadata)
 
+    def to_proto(
+        self, out: Optional[v2.device_pb2.DeviceSpecification] = None
+    ) -> v2.device_pb2.DeviceSpecification:
+        """Serializes the GridDevice to a DeviceSpecification.
+
+        Args:
+            out: Optional DeviceSpecification to be populated. Fields are populated in-place.
+
+        Returns:
+            The populated DeviceSpecification if out is specified, or the newly created
+            DeviceSpecification.
+        """
+        qubits = self._metadata.qubit_set
+        unordered_pairs = [tuple(pair_set) for pair_set in self._metadata.qubit_pairs]
```

Comment:
Should we sort self._metadata.qubit_pairs so that the order is stable.  I think qubit_pairs is just defined as an Iterable, so I am not sure if the ordering would be stable.

Reply:
True, added a sort to `pairs` below.